src/thunderbots/software/radio_communication/mrf/dongle.cpp

```diff
@@ -0,0 +1,610 @@
+
+    *rptr++ = static_cast<int8_t>(ballX);  // Add Ball x position
+    *rptr++ = static_cast<int8_t>(ballX >> 8);
+
+    *rptr++ = static_cast<int8_t>(ballY);  // Add Ball Y position
+    *rptr++ = static_cast<int8_t>(ballY >> 8);
+    struct
+    {
+        bool operator()(std::tuple<uint8_t, Point, Angle> a,
+                        std::tuple<uint8_t, Point, Angle> b) const
+        {
+            return std::get<0>(a) < std::get<0>(b);
+        }
+    } customLess;
+
+    std::sort(detbots.begin(), detbots.end(), customLess);
+
+    // For the number of robot for which data was passed in, assign robot ids to
+    // mask vector and position/angle data to camera packet
+    for (std::size_t i = 0; i < numbots; i++)
+    {
+        uint8_t robotID = std::get<0>(detbots[i]);
+        int16_t robotX  = static_cast<int16_t>((std::get<1>(detbots[i])).x() * 1000);
+        int16_t robotY  = static_cast<int16_t>((std::get<1>(detbots[i])).y() * 1000);
+        int16_t robotT =
+            static_cast<int16_t>((std::get<2>(detbots[i])).toRadians() * 1000);
+
+        mask_vec |= int8_t(0x01 << (robotID));
+        *rptr++ = static_cast<int8_t>(robotX);
+        *rptr++ = static_cast<int8_t>(robotX >> 8);
+        *rptr++ = static_cast<int8_t>(robotY);
+        *rptr++ = static_cast<int8_t>(robotY >> 8);
+        *rptr++ = static_cast<int8_t>(robotT);
+        *rptr++ = static_cast<int8_t>(robotT >> 8);
+    }
+
+    // Write out the timestamp
+    for (std::size_t i = 0; i < 8; i++)
+    {
+        *rptr++ = static_cast<int8_t>(timestamp >> 8 * i);
+    }
+
+    // Mask and Flag Vectors should be fully initialized by now. Assign them to
+    // the packet
+    camera_packet[0] = mask_vec;
+
+    std::lock_guard<std::mutex> lock(cam_mtx);
+
+    if (camera_transfers.size() >= 8)
+    {
+        LOG(WARNING) << "Camera transfer queue is full, ignoring camera packet"
+                     << std::endl;
+        return;
+    }
+
+    std::chrono::system_clock::time_point now = std::chrono::system_clock::now();
+    std::chrono::system_clock::time_point epoch =
+        std::chrono::system_clock::from_time_t(0);
+    std::chrono::system_clock::duration diff = now - epoch;
+    std::chrono::microseconds micros =
+        std::chrono::duration_cast<std::chrono::microseconds>(diff);
+    uint64_t stamp = static_cast<uint64_t>(micros.count());
+    std::unique_ptr<USB::BulkOutTransfer> elt(
+        new USB::BulkOutTransfer(device, 2, camera_packet, 55, 55, 0));
+    auto i = camera_transfers.insert(
+        camera_transfers.end(),
+        std::pair<std::unique_ptr<USB::BulkOutTransfer>, uint64_t>(std::move(elt),
+                                                                   stamp));
+    (*i).first->signal_done.connect(
+        sigc::bind(sigc::mem_fun(this, &MRFDongle::handle_camera_transfer_done), i));
+    (*i).first->submit();
+
+    LOG(INFO) << "Submitted camera transfer in position:" << camera_transfers.size()
```

Comment:
Could we move this to `LOG(DEBUG)`? I can't see much of a reason outside of debugging for this message, and we should try to keep "higher level" logs clean